graviti/schema/pyarrow.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+"""The PyArrow extension types to represent PortexType."""
+
+import json
+from typing import Any, Dict, Optional, Type, TypeVar
+
+import pyarrow as pa
+from pyarrow.lib import DataType  # pylint: disable=no-name-in-module
+
+# https://pylint.pycqa.org/en/latest/technical_reference/c_extensions.html
+
+
+class BuiltinExtension(  # pylint: disable=too-few-public-methods
+    pa.ExtensionType  # type: ignore[misc]
+):
+    """This class defines the PyArrow representation of PortexBuiltinType builtin types.
+
+    Arguments:
+        portex_type_name: The PortexBuiltinType name.
+        nullable: Whether the PortexBuiltinType is nullable.
+        storage_type: The corresponding PyArrow type.
+        kwargs: Other keyword arguments to init the PortexBuiltinType.
+
+    """
+
+    _T = TypeVar("_T", bound="BuiltinExtension")
+
+    def __init__(
+        self,
+        portex_type_name: str,
+        storage_type: DataType,
+        *,
+        nullable: bool = False,
```

Comment:
pyarrow have `nullable` parameter? can we reuse it?

Reply:
Only `pyarrow.field` have the `nullable` parameter.